src/amo/components/LandingPage/styles.scss

```diff
@@ -31,6 +27,8 @@
   height: 89px;
   width: 225px;
 
+  @include margin-end(20px);
```

Comment:
As mentioned in the issue, in LTR the spacing actually looks correct because of the (mostly) whitespace in the icon itself. If you zoom in you can see there is a faint squiggle:

<img width="125" alt="screenshot 2017-07-06 19 18 28" src="https://user-images.githubusercontent.com/90871/27926320-ebeba274-627f-11e7-90f5-83c4c551b34f.png">

This is rare case where we want separate styles for RTL, I think. So the themes spacing should return and `margin-left: 20px;` should be applied to `.LandingPage-icon`.